### PR TITLE
SAK-45116 MS Teams configuration checks

### PR DESCRIPTION
**Pull request by ropemar**
This PR includes a few changes to control exceptions when there's no MS Teams configuration on Sakai properties.

### CODE REVIEW

File: meetings/teams/src/java/org/sakaiproject/meetings/teams/MicrosoftTeamsService.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import com.microsoft.graph.requests.GroupCollectionRequestBuilder;
 import com.microsoft.graph.requests.UserCollectionPage;
 import com.microsoft.graph.requests.UserCollectionRequestBuilder;
+import com.nimbusds.oauth2.sdk.util.StringUtils;
```
Comment (mpellicer) [inline meetings/teams/src/java/org/sakaiproject/meetings/teams/MicrosoftTeamsService.java:14]: I think we should use Apache's StringUtils

File: meetings/teams/src/java/org/sakaiproject/meetings/teams/MicrosoftTeamsService.java [hunk 2]
```diff
@@ -67,6 +68,22 @@ public MicrosoftTeamsService() {
...
+     * Checking that MSTeams has been set up
+     * @return
+     */
+    public boolean isMicrosofTeamsConfigured() {
+        boolean result = false;
+        String authority = serverConfigurationService.getString(MSTEAMS_PREFIX + AUTHORITY, null);
+        String clientId = serverConfigurationService.getString(MSTEAMS_PREFIX + CLIENT_ID, null);
+        String secret = serverConfigurationService.getString(MSTEAMS_PREFIX + SECRET, null);
+        String scope = serverConfigurationService.getString(MSTEAMS_PREFIX + SCOPE, null);
+        if (StringUtils.isNotBlank(authority) && StringUtils.isNotBlank(clientId) && StringUtils.isNotBlank(secret) && StringUtils.isNotBlank(scope)) {
```
Comment (mpellicer) [inline meetings/teams/src/java/org/sakaiproject/meetings/teams/MicrosoftTeamsService.java:81]: return StringUtils.isNoneBlank(authority, clientId, secret, scope) !!

File: meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java [hunk 3]
```diff
@@ -606,6 +606,16 @@ public String getI18nProperties(@PathVariable String bundle, @PathVariable Strin
         return i18n.toString();
     }
     
+    /**
+     * Returns true if MS Teams is set up in Sakai properties
+     * @return
+     * @throws MeetingsException
+     */
+    @GetMapping(value = "/meetings/isteamssetup", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (mpellicer) [inline meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java:614]: I'm not an expert on API definition but maybe /meetings/teams/status is better?

File: meetings/tool/src/main/resources/create-meeting.properties [hunk 4]
```diff
@@ -5,6 +5,7 @@ cancel=Cancel
 close_date=Close Date
 error_create_meeting_500=There was a problem creating the meeting
 error_create_meeting_unknown=Unknown error when trying to create meeting. Response code:
+error_video_conferencing_config=There is no configuration of the video conferencing service. Please check this with your Sakai administrator.
```
Comment (mpellicer) [inline meetings/tool/src/main/resources/create-meeting.properties:8]: The meetings tool requires at least one provider like Microsoft Teams or Zoom, please contact your service administrator or vendor.

File: meetings/tool/src/main/resources/create-meeting_es.properties [hunk 5]
```diff
@@ -5,6 +5,7 @@ cancel=Cancelar
 close_date=Fecha de cierre
 error_create_meeting_500=Hubo un problema al crear la reuni\u00F3n
 error_create_meeting_unknown=Error desconocido al intentar crear la reuni\u00F3n. C\u00F3digo de respuesta:
+error_video_conferencing_config=No hay configuraci\u00F3n del servicio de videoconferencia. Por favor, compruebe el problema con su administrador de Sakai.
```
Comment (mpellicer) [inline meetings/tool/src/main/resources/create-meeting_es.properties:8]: La herramienta necesita un proveedor como Microsoft Meetings o Zoom. Por favor contacta con el administrador o el proveedor del servicio.

**Review by mpellicer — CHANGES_REQUESTED** [review]

File: meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java (hunk 3 above)
Comment (ropemar) [reply meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java:614]: Yeah, that makes more sense to me, I'm gonna change it

File: meetings/teams/src/java/org/sakaiproject/meetings/teams/MicrosoftTeamsService.java [hunk 6]
```diff
@@ -67,6 +69,18 @@ public MicrosoftTeamsService() {
...
+    /**
+     * Checking that MSTeams has been set up
+     * @return
+     */
+    public boolean isMicrosofTeamsConfigured() {
+        String authority = serverConfigurationService.getString(MSTEAMS_PREFIX + AUTHORITY, null);
+        String clientId = serverConfigurationService.getString(MSTEAMS_PREFIX + CLIENT_ID, null);
+        String secret = serverConfigurationService.getString(MSTEAMS_PREFIX + SECRET, null);
+        String scope = serverConfigurationService.getString(MSTEAMS_PREFIX + SCOPE, null);
+        return StringUtils.isNoneBlank(authority, clientId, secret, scope);
```
Comment (mpellicer) [inline meetings/teams/src/java/org/sakaiproject/meetings/teams/MicrosoftTeamsService.java:81]: NICE!

**Review by mpellicer — APPROVED** [review]
Thanks for the changes

**Comment by ropemar** [comment]
You're welcome, @mpellicer !

**Pull request merged**